api/views.py

```diff
@@ -393,17 +391,19 @@ def aggregate_general_content(self, report_object, shifts):
         content["long_month_name"] = month_names[report_object.month_year.month]
 
         # Data for Footer
-        content["debit_work_time"] = "{hours:02g}:{minutes:02g}".format(
-            hours=contract_hours, minutes=60 * round(contract_hours_decimal, 2)
+        td = datetime.timedelta(minutes=report_object.contract.minutes)
+        hours_minutes = str(td)[:-3].split(":")
+        content["debit_work_time"] = ":".join(
```

Comment:
Waruuuum? Das ist so hässlich und ich verstehe nicht mal beim nachdenken was diese zwei Zeilen machen. Was war an der vorherigen version schlimm?

Reply:
Ich finde die `modf` Lösung undurchsichtiger. Ich bin mit der Syntax hier auch nicht 100% zufrieden, aber es ist wie folgt:

`str(datetime.timedelta())` gibt `6:00:00` aus für 6 Stunden, 0 Minuten und 0 Sekunden. Der String Splice (`[:-3]`) selektriert nur den Stunden-Minuten-Teil (`6:00`). Mit dem Split erhalten wir die einzelnen Objekte und die List Comprehesion verwandelt diese in Integer, um sie dann in mit 0-gepaddete Strings umzuwandeln.

Ich denke noch mal darüber nach wie man das verbessern kann, will aber die `modf` Nutzung vermeiden.

Reply:
Es kann aber ziemlich gut sein, dass das hier nicht error-proof ist. Eventuell stürzt das Backend ab, wenn der Timedelta >24h ist, da der Slice+Split nicht funktionieren wird.

Reply:
Dann kannst du aber genauso gut,
```
td = relativedelta(minutes=report_object.contract.minutes)
content["debit_work_time"] = relativedelta_to_string(td)
```
machen. `relativedelta_to_string` ist eine eigens geschriebene utility Funktion.

Reply:
Ach um so besser! Dann ändere ich das so